### PR TITLE
Related to #176. Ensure IdP certificate is formatted properly

### PR DESCRIPTION
**Pull request by pitbulk**
PR #176 + Rebase

### CODE REVIEW

File: lib/onelogin/ruby-saml/utils.rb [hunk 1]
```diff
@@ -4,58 +4,40 @@ module RubySaml
...
       # @return [String] The formatted certificate
       #
-      def self.format_cert(cert, heads=true)
-        if cert && !cert.empty?
-          cert = cert.gsub(/\-{5}\s?(BEGIN|END) CERTIFICATE\s?\-{5}/, "")
-          cert = cert.gsub(/[\n\r\s]/, "")
-          cert = cert.scan(/.{1,64}/).join("\n")+"\n"
+      def self.format_cert(cert)
+        # don't try to format an encoded certificate or if is empty or nil
+        return cert if cert.nil? || cert.empty? || cert.match(/\x0d/)
```
Comment (luisvm) [inline lib/onelogin/ruby-saml/utils.rb]: I don't think this has the right logic, can we review?

Comment (luisvm) [inline lib/onelogin/ruby-saml/utils.rb]: oh! it is, it just looked weird to me... disregard please

**Comment by luisvm** [comment]
:+1: looks good

**Comment by pitbulk** [comment]
If I try to get info from the cert
```
openssl x509 -in valid.crt -text -noout
```
I got:
```
unable to load certificate
139721105524416:error:0D07207B:asn1 encoding routines:ASN1_get_object:header too long:asn1_lib.c:150:
139721105524416:error:0D068066:asn1 encoding routines:ASN1_CHECK_TLEN:bad object header:tasn_dec.c:1306:
139721105524416:error:0D08303A:asn1 encoding routines:ASN1_TEMPLATE_NOEXP_D2I:nested asn1 error:tasn_dec.c:659:
139721105524416:error:0D08303A:asn1 encoding routines:ASN1_TEMPLATE_NOEXP_D2I:nested asn1 error:tasn_dec.c:751:Field=subject, Type=X509_CINF
139721105524416:error:0D08303A:asn1 encoding routines:ASN1_TEMPLATE_NOEXP_D2I:nested asn1 error:tasn_dec.c:751:Field=cert_info, Type=X509
139721105524416:error:0906700D:PEM routines:PEM_ASN1_read_bio:ASN1 lib:pem_oth.c:83:
```
So definitely this certificate is not valid...

**Comment by pitbulk** [comment]
Same happens with the private_key

```
openssl rsa -in valid_private_key -check
```
```
unable to load Private Key
140452807743168:error:0D0680A8:asn1 encoding routines:ASN1_CHECK_TLEN:wrong tag:tasn_dec.c:1319:
140452807743168:error:0D07803A:asn1 encoding routines:ASN1_ITEM_EX_D2I:nested asn1 error:tasn_dec.c:381:Type=X509_ALGOR
140452807743168:error:0D08303A:asn1 encoding routines:ASN1_TEMPLATE_NOEXP_D2I:nested asn1 error:tasn_dec.c:751:Field=pkeyalg, Type=PKCS8_PRIV_KEY_INFO
140452807743168:error:0907B00D:PEM routines:PEM_READ_BIO_PRIVATEKEY:ASN1 lib:pem_pkey.c:132:
```

**Pull request merged**